HadesScheduler/k8s/k8s.go

```diff
@@ -92,7 +145,12 @@ func initializeClusterAccess(k8sCfg K8sConfig) Scheduler {
 }
 
 func (k Scheduler) ScheduleJob(ctx context.Context, job payload.QueuePayload) error {
-	slog.Debug("Scheduling job in Kubernetes")
+	if k.useOperator {
```

Comment:
I dont think this is a good idea. The `K8sCfg` scruct has a a config mode. I'd rather use that to select the scheduling method. Please also extract each scheduling option into extra functions similar to the `createBuildJobCR`